### PR TITLE
[stable/mysqldump] Add support for uploading backups to OpenStack

### PR DESCRIPTION
**Pull request by marcus-sonestedt**
Signed-off-by: Marcus Sonestedt <marcus.s.lindblom@gmail.com>

#### What this PR does / why we need it:

This adds support for uploading backups to OpenStack's objectstore, via explicit password or existing secret/key combination.

Also, a time-to-live can be set which expires the files automatically, similar to the current housekeeping.

**Additional features**

* Does not require a PVC to run anymore. That is not needed when the backups are uploaded somewhere else, methinks. Previously, the whole backup stage was skipped if there was no persistent volume, which seemed a bit odd.
* Supports setting security context so the scripts do not run as root. Disabled by default to avoid surprises.

#### Which issue this PR fixes

n/a

#### Special notes for your reviewer:

Python2 is present in current image, so I wrote a script that calls on curl to do the heavy lifting, which made it easier to copy working examples that used bash/curl. (json manipulation with bash scripting is not my strongest side...)

@paulczar @monotek @axdotl

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by marcusl** [comment]
Here's a sample log output:

```test mysql connection
Backing up db keycloak
Backing up db mysql
Backing up db test
upload files to openstack at backup/test.barium.cloud/full/mariadb/idb
This script uploads files to an OpenStack object store container

Getting authentication token from https://ops.elastx.cloud:5000/v3/auth/tokens ...
Locating public object store URL in catalog ...
  https://swift.elastx.cloud:443/v1/AUTH_ed69fa4ea47a4086a85cf6e994b298c4
Checking container: backup ...
  HTTP/1.1 204 No Content

Uploading
  from: /backup
  to:   https://swift.elastx.cloud:443/v1/AUTH_ed69fa4ea47a4086a85cf6e994b298c4/backup/test.barium.cloud/full/mariadb/idb
  ttl:  7 day(s)

  /backup/20190813090306.state
    Uploading ...
  /backup/keycloak/20190813090306_keycloak.sql.gz
    Uploading ...
  /backup/test/20190813090306_test.sql.gz
    Uploading ...
  /backup/mysql/20190813090306_mysql.sql.gz
    Uploading ...

Uploaded 4 of 4 file(s), 0 failure(s)
Disk usage in /backup
60.0K   /backup/keycloak
8.0K    /backup/test
148.0K  /backup/mysql
224.0K  /backup
Backup successful! :-)```

**Comment by marcusl** [comment]
Here's a sample log output:

```test mysql connection
Backing up db keycloak
Backing up db mysql
Backing up db test
upload files to openstack at backup/test.xxxx.cloud/full/mariadb/idb
This script uploads files to an OpenStack object store container

Getting authentication token from https://ops.elastx.cloud:5000/v3/auth/tokens ...
Locating public object store URL in catalog ...
  https://swift.elastx.cloud:443/v1/AUTH_xxxxxxxxxxxxxxxx
Checking container: backup ...
  HTTP/1.1 204 No Content

Uploading
  from: /backup
  to:   https://swift.elastx.cloud:443/v1/AUTH_xxxxxxxxxxxxxxxx/backup/test.xxxx.cloud/full/mariadb/idb
  ttl:  7 day(s)

  /backup/20190813090306.state
    Uploading ...
  /backup/keycloak/20190813090306_keycloak.sql.gz
    Uploading ...
  /backup/test/20190813090306_test.sql.gz
    Uploading ...
  /backup/mysql/20190813090306_mysql.sql.gz
    Uploading ...

Uploaded 4 of 4 file(s), 0 failure(s)
Disk usage in /backup
60.0K   /backup/keycloak
8.0K    /backup/test
148.0K  /backup/mysql
224.0K  /backup
Backup successful! :-)```

File: stable/mysqldump/templates/configmap.yaml [hunk 1]
```diff
@@ -10,6 +10,16 @@ data:
...
   MYSQL_OPTS: {{ .Values.options | quote }}
   KEEP_DAYS: {{ .Values.housekeeping.keepDays | quote }}
+{{- if .Values.upload.openstack.enabled }}
+  OS_AUTH_URL: {{ .Values.upload.openstack.authUrl }}
+  OS_PROJECT_NAME: {{ .Values.upload.openstack.project }}
+  OS_PROJECT_DOMAIN_NAME: {{ .Values.upload.openstack.projectDomain }}
+  OS_USERNAME: {{ .Values.upload.openstack.user }}
+  OS_USER_DOMAIN_NAME: {{ .Values.upload.openstack.userDomain }}
+{{- if not .Values.upload.openstack.existingSecret }}
+  OS_PASSWORD: {{ .Values.upload.openstack.password }}
```
Comment (monotek) [inline stable/mysqldump/templates/configmap.yaml]: this should be referenced from a secret.

**Review by monotek — CHANGES_REQUESTED** [review]

File: stable/mysqldump/templates/configmap.yaml (hunk 1 above)
Comment (marcus-sonestedt) [reply stable/mysqldump/templates/configmap.yaml]: True. Thanks for spotting it.

**Review by monotek — APPROVED** [review]
/lgtm

**Pull request merged**